components/gfx/platform/macos/font.rs

```diff
@@ -155,12 +245,38 @@ impl FontHandleMethods for FontHandle {
         return Some(glyphs[0] as GlyphId);
     }
 
-    fn glyph_h_kerning(&self, _first_glyph: GlyphId, _second_glyph: GlyphId)
-                        -> FractionalPixel {
-        // TODO: Implement on mac
+    fn glyph_h_kerning(&self, first_glyph: GlyphId, second_glyph: GlyphId) -> FractionalPixel {
+        let table = match self.h_kern_subtable {
+            Some(ref table) => table,
+            None => return 0.0
+        };
+
+        // Binary search.
+        let pairs = table.pair_data();
+        let query = first_glyph << 16 | second_glyph;
+        let (mut start, mut end) = (0, pairs.len() / KERN_PAIR_LEN);
+
+        while start < end {
+            let i = (start + end) / 2;
+            let key = BigEndian::read_u32(&pairs[i * KERN_PAIR_LEN..]);
+            if key > query {
+                end = i;
+            } else if key < query {
+                start = i + 1;
+            } else {
+                let value = BigEndian::read_i16(&pairs[i * KERN_PAIR_LEN + 4..]);
+                trace!("Kerning for {}/{}: {:?}", first_glyph, second_glyph, value);
+                return value as f64 * table.scale;
+            }
```

Comment:
Can this use `Vec::binary_search()`?

Even if not, I still think I'd rather have this be a separate function.